ambry-tools/src/main/java/com.github.ambry/store/DumpData.java

```diff
@@ -562,40 +559,31 @@ public void compareReplicaIndexEntriestoLogContent(String replicaRootDirectory,
     RandomAccessFile randomAccessFile = new RandomAccessFile(new File(logFile), "r");
     logger.info("Comparing Index entries to Log ");
     File replicaDirectory = new File(replicaRootDirectory);
-    MergedIntervals coveredRangesInLog = new MergedIntervals();
-    Map<Long, Long> deletedMsgInfo = new HashMap<>();
     for (File indexFile : replicaDirectory.listFiles(new IndexFileNameFilter())) {
-      compareIndexEntriesToLogContent(indexFile, replicaDirectory, randomAccessFile, coveredRangesInLog, deletedMsgInfo,
-          false);
+      TreeMap<Long, Long> coveredRanges = new TreeMap<Long, Long>();
+      Set<Long> deleteMsgsOriginalOffsets = new HashSet<Long>();
+      compareIndexEntriesToLogContent(indexFile, replicaDirectory, randomAccessFile, coveredRanges,
+          deleteMsgsOriginalOffsets);
     }
```

Comment:
We can do our best in trying to get the first offset of first segment and end offset of last segment. But still we might have to parse the logs to know the holes at the end. But to the minimum, we can do what index consistency checker does if you feel strongly.